src/main/java/org/assertj/core/internal/Objects.java

```diff
@@ -651,6 +653,23 @@ private boolean isActualIn(Object actual, Iterable<?> values) {
   }
 
   /**
+   * Assert that the given object is "deeply" equals to the other by comparing all fields recursively.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the given object.
+   * @param other the object to compare {@code actual} to.
+   * @throws AssertionError if actual is {@code null}.
+   * @throws AssertionError if the other object is not an instance of the actual type.
+   * @throws AssertionError if the actual and the given object are not "deeply" equal.
+   */
+  public <A> void assertIsEqualToComparingFieldByFieldRecursively(AssertionInfo info, A actual, A other) {
+    assertNotNull(info, actual);
+    assertHasSameClassAs(info, actual, other);
+    if (!DeepEquals.deepEquals(actual, other))
+        throw failures.failure(info, shouldBeEqual(actual, other, StandardRepresentation.STANDARD_REPRESENTATION));
```

Comment:
Sorry for the late feedback :(
It seems `deepEquals` only helps partially, not considering properties is a blocking issue as we want to have a consistent behaviour between whether the field by field comparison is recursive or not.
Moreover we need an implementation that tracks field/property path, if we only report that objects are not equal but not show why it would not be helpful for users.

I will give it a shot by the end of the year, first by looking for other `deep` equals implementation.
If you want to continue working on this one, feel free to do so, your work is always appreciated.